### ISSUE TITLE
`superfluous-parens` false positive for `is not` tests

### ISSUE THREAD

**Issue by ryansdowning** [PyCQA/pylint#5930]
### Bug description

I just got the lint `C0325: Unnecessary parens after 'not' keyword (superfluous-parens)`, so I removed the parens it referred to, but thankfully a unit test caught that these are not the same, and the parens are needed (removing them changes the logic!).

I have been able to create a minimal example that I believe reproduces this

```python
In [1]: (None is None) is not (None == 1)  # Original, pylint lints with C0325
Out[1]: True

In [1]: (None is None) is not None == 1  # Recommended by C0325
Out[1]: False
```

### Command used

```shell
pylint path/to/pkg/ --max-line-length=120 --fail-under=9 --load-plugins pylint_django
```

### Pylint output

```shell
************* Module pkg.module
path/to/pkg/module.py:63:0: C0325: Unnecessary parens after 'not' keyword (superfluous-parens)
```

### Expected behavior

C0325 should not be output here.

### Pylint version

```shell
pylint 2.11.1
astroid 2.8.6
Python 3.9.7 (default, Sep 16 2021, 13:09:58) 
[GCC 7.5.0]
```

**Comment by jacobtylerwalls** [comment]
Thanks for the report. This is _not_ a duplicate of #5803 (my diff for that doesn't fix this!)

There are numerous operators that aren't working. At a glance, it looks like the implementation of the check confuses `is not` with `not`, which have different [precedences](https://docs.python.org/3/reference/expressions.html#operator-precedence). That would explain why the operators in the same group as `is not` produce false positives:
```python
x is not (x <= y)
True is not (True == False)
True is not (True is not False)
```

**Comment by timmartin** [comment]
If I understand correctly, the problem isn't so much that `not` and `is not` have different precedence (they are different, but no other operator is between them), but that the formatting checker assumes that a `not` necessarily indicates a unary operator when in fact it can be part of a binary operator expression in the form of `is not`.
